pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/columnminmaxvalue/ColumnMinMaxValueGenerator.java

```diff
@@ -124,227 +131,251 @@ private List<String> getColumnsToAddMinMaxValue() {
   }
 
   private boolean needAddColumnMinMaxValueForColumn(String columnName) {
-    ColumnMetadata columnMetadata = _segmentMetadata.getColumnMetadataFor(columnName);
+    return needAddColumnMinMaxValueForColumn(_segmentMetadata.getColumnMetadataFor(columnName));
+  }
+
+  private boolean needAddColumnMinMaxValueForColumn(ColumnMetadata columnMetadata) {
     return columnMetadata.getMinValue() == null && columnMetadata.getMaxValue() == null
         && !columnMetadata.isMinMaxValueInvalid();
   }
 
-  private void addColumnMinMaxValueForColumn(String columnName)
-      throws Exception {
-    // Skip column with min/max value already set
+  private void addColumnMinMaxValueForColumn(String columnName) {
     ColumnMetadata columnMetadata = _segmentMetadata.getColumnMetadataFor(columnName);
-    if (columnMetadata.getMinValue() != null || columnMetadata.getMaxValue() != null) {
+    if (!needAddColumnMinMaxValueForColumn(columnMetadata)) {
       return;
     }
+    try {
+      if (columnMetadata.hasDictionary()) {
+        addColumnMinMaxValueWithDictionary(columnMetadata);
+      } else {
+        addColumnMinMaxValueWithoutDictionary(columnMetadata);
+      }
+      _minMaxValueAdded = true;
+    } catch (Exception e) {
+      LOGGER.error("Caught exception while generating min/max value for column: {} in segment: {}, continuing without "
```

Comment:
this is the part that's definitely more informed and thorough than my fix. So any errors here are now logged rather than throwing. thank you